src/providers/microsoft/azure/mod.rs

```diff
@@ -271,10 +271,14 @@ impl Azure {
             .context("failed to decrypt cms blob")?;
 
         // convert that to the OpenSSH public key format
-        let ssh_pubkey = crypto::p12_to_ssh_pubkey(&p12)
-            .context("failed to convert pkcs12 blob to ssh pubkey")?;
+        let ssh_pubkey = match crypto::p12_to_ssh_pubkey(&p12)
+            .context("failed to convert pkcs12 blob to ssh pubkey")?
+        {
+            Some(key) => key,
+            None => return Ok(None),
+        };
```

Comment:
What I mean is that all this can now be

```suggestion
        crypto::p12_to_ssh_pubkey(&p12)
            .context("failed to convert pkcs12 blob to ssh pubkey")
```

And we can remove the `Ok()` line below.

Reply:
Updated! Thank you.

Reply:
I didnt update it properly. I'll work on it.